android/src/main/java/io/ably/lib/push/ActivationContext.java

```diff
@@ -95,21 +95,21 @@ public boolean setClientId(String clientId, boolean propagateGotPushDeviceDetail
         return updated;
     }
 
-    public void onNewRegistrationToken(RegistrationToken.Type type, String token) {
-        Log.v(TAG, "onNewRegistrationToken(): type=" + type + ", token=" + token);
+    public void onNewRegistrationToken(String token) {
+        Log.v(TAG, "onNewRegistrationToken(): type=" + RegistrationToken.Type.FCM + ", token=" + token);
         LocalDevice localDevice = getLocalDevice();
         RegistrationToken previous = localDevice.getRegistrationToken();
         if (previous != null) {
-            if (previous.type != type) {
-                Log.e(TAG, "trying to register device with " + type + ", but it was already registered with " + previous.type);
+            if (previous.type != RegistrationToken.Type.FCM) {
+                Log.e(TAG, "trying to register device with " + RegistrationToken.Type.FCM + ", but it was already registered with " + previous.type);
```

Comment:
If this happens, what can a user do to resolve it? Or perhaps instead of doing nothing (just returning), we could call `localDevice.setAndPersistRegistrationToken(new RegistrationToken(token));`.

Reply:
yes sure, if that's appropriate handling

Reply:
This would have to be validated to work. Would realtime servers be updated in response to the new token? Looking at the code, I don't think so. Therefore, just removing the `return` statement will not be a good idea? What do you think?